pkg/cmd/roachtest/tests/failure_injection.go

```diff
@@ -0,0 +1,272 @@
+) error {
+	if err := t.validateFailure(ctx, l, c); err == nil {
+		return errors.New("no failure was injected but validation still passed")
+	}
+	if err := t.validateRestore(ctx, l, c); err != nil {
+		return errors.Wrapf(err, "no failure was injected but post restore validation still failed")
+	}
+	return nil
+}
+
+var bidirectionalNetworkPartitionTest = failureSmokeTest{
+	testName:    "bidirectional network partition",
+	failureName: failures.IPTablesNetworkPartitionName,
+	args: failures.NetworkPartitionArgs{
+		Partitions: []failures.NetworkPartition{{
+			Source:      install.Nodes{1},
+			Destination: install.Nodes{3},
+			Type:        failures.Bidirectional,
+		}},
+	},
+	waitForFailureToPropagate: 5 * time.Second,
+	validateFailure: func(ctx context.Context, l *logger.Logger, c cluster.Cluster) error {
+		blocked, err := checkPortBlocked(ctx, l, c, c.Nodes(1), c.Nodes(3))
+		if err != nil {
+			return err
+		}
+		if !blocked {
+			return fmt.Errorf("expected connections from node 1 to node 3 to be blocked")
+		}
+
+		blocked, err = checkPortBlocked(ctx, l, c, c.Nodes(1), c.Nodes(2))
+		if err != nil {
+			return err
+		}
+		if blocked {
+			return fmt.Errorf("expected connections from node 1 to node 2 to not be blocked")
+		}
+		return nil
+	},
+	validateRestore: func(ctx context.Context, l *logger.Logger, c cluster.Cluster) error {
+		blocked, err := checkPortBlocked(ctx, l, c, c.Nodes(1), c.Nodes(3))
+		if err != nil {
+			return err
+		}
+		if blocked {
+			return fmt.Errorf("expected connections from node 1 to node 3 to not be blocked")
+		}
+		return err
+	},
+}
+
+var asymmetricNetworkPartitionTest = failureSmokeTest{
+	testName:    "asymmetric network partition",
+	failureName: failures.IPTablesNetworkPartitionName,
+	args: failures.NetworkPartitionArgs{
+		Partitions: []failures.NetworkPartition{{
+			Source:      install.Nodes{1},
+			Destination: install.Nodes{3},
+			Type:        failures.Incoming,
+		}},
+	},
+	waitForFailureToPropagate: 5 * time.Second,
+	validateFailure: func(ctx context.Context, l *logger.Logger, c cluster.Cluster) error {
+		blocked, err := checkPortBlocked(ctx, l, c, c.Nodes(1), c.Nodes(3))
+		if err != nil {
+			return err
+		}
+		if blocked {
+			return fmt.Errorf("expected connections from node 1 to node 3 to be open")
+		}
+
+		blocked, err = checkPortBlocked(ctx, l, c, c.Nodes(3), c.Nodes(1))
+		if err != nil {
+			return err
+		}
+		if !blocked {
+			return fmt.Errorf("expected connections from node 3 to node 1 to be blocked")
+		}
+		return nil
+	},
+	validateRestore: func(ctx context.Context, l *logger.Logger, c cluster.Cluster) error {
+		blocked, err := checkPortBlocked(ctx, l, c, c.Nodes(1), c.Nodes(3))
+		if err != nil {
+			return err
+		}
+		if blocked {
+			return fmt.Errorf("expected connections from node 1 to node 3 to not be blocked")
+		}
+		return err
+	},
+}
+
+// Helper function that uses nmap to check if connections between two nodes is blocked.
+func checkPortBlocked(
```

Comment:
This could be lifted to a (lib) helper if it also takes the port arg.

Reply:
Done